### PR TITLE
Feature/m2 2342 subscale calc

### PR DESCRIPTION
**Pull request by kdanilenka32**

### CODE REVIEW

File: src/shared/utils/exportData/getSubscales.ts [hunk 1]
```diff
@@ -1,11 +1,140 @@
...
-    acc[item.name] = '1';
-    if (item?.subscaleTableData) {
-      acc[`Optional text for ${item.name}`] = item.subscaleTableData;
+import { getObjectFromList } from '../builderHelpers';
+import { createArrayFromMinToMax } from '../array';
+
+export const getSubScaleScore = (subscalesSum: number, type: SubscaleTotalScore, length: number) =>
+  type === SubscaleTotalScore.Sum ? subscalesSum : subscalesSum / length;
+
+export const parseSex = (sex: string) => (sex === 'M' ? '1' : '2');
```
Comment (alex-i-e) [inline src/shared/utils/exportData/getSubscales.ts:26]: Please add enum for 'M' / 'W'

File: src/shared/utils/exportData/prepareData.ts [hunk 2]
```diff
@@ -41,7 +42,16 @@ export const prepareData = (
           }),
         );
       }, [] as ReturnType<typeof getReportCSVObject>[]);
-      const reportData = acc.reportData.concat(...answers);
+
+      const subscaleSetting = data.decryptedAnswers?.[0]?.subscaleSetting;
```
Comment (alex-i-e) [inline src/shared/utils/exportData/prepareData.ts:46]: You added subscales calc only for first answer. Can you confirm it should be skipped for others?

Comment (kdanilenka32) [reply src/shared/utils/exportData/prepareData.ts:46]: Yes

File: src/shared/utils/exportData/getSubscales.ts [hunk 3]
```diff
@@ -1,11 +1,140 @@
...
+        scoring: subscaleSetting.calculateTotalScore,
+        subscaleTableData: subscaleSetting.totalScoresTableData,
+      } as ActivitySettingsSubscale,
+      activityItems,
+      {},
+    );
+
+  return {
+    ...(calculetedTotalScore && {
+      'Final SubScale Score': calculetedTotalScore.score,
```
Comment (wildkiomi) [inline src/shared/utils/exportData/getSubscales.ts:135]: it shouldn't be translated?

**Review by wildkiomi — APPROVED** [review]

File: src/shared/utils/exportData/getSubscales.ts [hunk 4]
```diff
@@ -1,11 +1,140 @@
...
+    }
+
+    return acc + value;
+  }, 0);
+
+  const calculetedScore = getSubScaleScore(sumScore, data.scoring, data.items.length);
+
+  if (data?.subscaleTableData) {
+    const subscaleTableDataItem = data.subscaleTableData?.find(({ sex, age, rawScore }) => {
+      const genderAnswer = activityItems.gender_screen?.answer as DecryptedSexAnswer;
```
Comment (alex-i-e) [inline src/shared/utils/exportData/getSubscales.ts:82]: please add enum for 'gender_screen' and `age_screen` and use it for `name` properties in genderItem and ageItem in src/modules/Builder/features/ActivitySettings/SubscalesConfiguration/SubscalesConfiguration.const.ts

**Review by alex-i-e — APPROVED** [review]

**Pull request merged**